src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -31,7 +31,8 @@
           if (logger.isInfoEnabled()) {
             logger.info("Successfully retrieved invoice lines: {}", jsonInvoiceLines.encodePrettily());
           }
-          future.complete(jsonInvoiceLines.mapTo(InvoiceLineCollection.class));
+          VertxCompletableFuture.supplyBlockingAsync(ctx, () -> jsonInvoiceLines.mapTo(InvoiceLineCollection.class))
```

Comment:
In case `mapTo` operation fails with exception, `future` will never be completed. Probably you can change to something like

```java
  .thenCompose(jsonInvoiceLines -> VertxCompletableFuture.supplyBlockingAsync(ctx, () -> {
    if (logger.isInfoEnabled()) {
      logger.info("Successfully retrieved invoice lines: {}", jsonInvoiceLines.encodePrettily());
    }
    return jsonInvoiceLines.mapTo(InvoiceLineCollection.class);
  }))
  .thenAccept(future::complete)
  .exceptionally(t -> {
  ...
```